### ISSUE TITLE
Window/grid specific font size

### ISSUE THREAD

**Issue by IndianBoy42** [neovide/neovide#1874]
**Is your feature request related to a problem? Please describe.**
Not so much a problem, just an opportunity to match vscode and other GUI editors. It can be useful to have different windows at different font sizes, either for focusing on different editing windows, or making sidebars/floating windows more compact. Maybe even the bufferline/statusline (I am not sure how much control `ext_multigrid` gets you)

**Describe the solution you'd like**
As much customization as is currently possible for window local font size as well as varying the font size for other elements of the GUI. As I understand it this is probably heavily dependent on how much neovims `ext_*` UI allows.

**Comment by Kethku** [comment]
There are two main issues that block this for now

1. For non floating windows, setting the font size means that the content must take up more or less space to compensate. The way this is supported today in neovim, that would require neovide to handle all of the grid positioning it self. We've carefully avoided this so far to have as much compatibility with the way neovim lays things out. The last thing we want is for neovide to display windows or interact with windows differently from the default app.
2. We dont currently have a way to observe window specific settings. This isn't an impossible task, but just isn't currently done yet. Its something I personally have wanted for a while though, and have made some small steps toward in https://github.com/neovide/neovide/pull/2111. I expect the way this would work is to observe window local options like guifont and adjust the font size or w/e from there.

So to summarize. Its not too far off to implement something like this soonish, but there are some open issues to figure out still. And to implement it well, we probably need some upstream changes.
